homeassistant/components/zeroconf/__init__.py

```diff
@@ -386,19 +369,17 @@ def info_from_service(service):
 
     address = service.addresses[0]
 
-    info = {
-        ATTR_HOST: str(ipaddress.ip_address(address)),
-        ATTR_PORT: service.port,
-        ATTR_HOSTNAME: service.server,
-        ATTR_TYPE: service.type,
-        ATTR_NAME: service.name,
-        ATTR_PROPERTIES: properties,
+    return {
+        "host": str(ipaddress.ip_address(address)),
```

Comment:
What was the reasoning behind removing the constants?

Reply:
This is now a `TypedDict`, it also supports string literals as keys, not variables. But mypy will check these keys are correctly spelled, all of them are set and they have the specified type. So this is actually safer than before and there is a single source of the truth in `HaServiceInfo` definition.

Reply:
`TypedDict` also allows to remove few check if key exists because mypy guarantees that.